lib/llvm/Support/Path.cpp

```diff
@@ -898,6 +898,9 @@ void directory_entry::replace_filename(const Twine &filename, file_status st) {
   Status = st;
 }
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wimplicit-fallthrough"
```

Comment:
Please use the fallthrough attribute here as well.

Reply:
My thought here with the pragma was that this was imported code that is making rather extensive use of the fallthrough in a way that readability will not be improved with the attributes.  We'll be diverging from the imported source fairly extensively.

I don't disagree strongly enough not to make the changes, but I did select this purposely.

Reply:
Alright, I am a bit confused at how clang processes this code. (previous comment retracted).  There is extensive use of fallthrough here, but it only warns on a single line that is exactly the same as other lines in this code.  I am rather dubious of the attribute use and warning processing here.

Reply:
Ugh, actually, it appears to be non-deterministic in its registration of the warnings?

Reply:
Yeah, there is really only 1 fall through. The multiple case labels should be referring to 1 block of code, which is not actually fall through. (update my understanding of how the code is processed). The oddity with the warning on my system notwithstanding, I'll post the change in a moment.